### PR TITLE
Form elements

### PR DESCRIPTION
**Pull request by bengammon**
The form elements branch is now in a good place for review and would be good to get a second set of eyes on it...

### CODE REVIEW

**Comment by peterkeating** [comment]
I'll come over and fix the build issue with you :)

File: content/Assets/Styles/components/form/_checkbox.scss [hunk 1]
```diff
@@ -0,0 +1,89 @@
+/* ==========================================================================
```
Comment (peterkeating) [inline content/Assets/Styles/components/form/_checkbox.scss]: This file should be named `checkboxes` to match the primary selector (comment should be updated to).

File: content/Assets/Styles/components/form/_radio.scss [hunk 2]
```diff
@@ -0,0 +1,90 @@
+/* ==========================================================================
```
Comment (peterkeating) [inline content/Assets/Styles/components/form/_radio.scss]: As per checkboxes, rename to "radios".

File: content/Assets/Styles/settings/_form.scss [hunk 3]
```diff
@@ -0,0 +1,26 @@
...
+$form-radio-height: 22px;
+$form-radio-width: 22px;
+
+$form-radio-label-padding-left: 30px;
+$form-radio-label-after-border-width: 6px;
+$form-radio-label-after-x-pos: 5px;
+$form-radio-label-after-y-pos: 5px;
+
+$form-element-padding-horizontal: 5px;
+$form-element-padding-vertical: 5px;
```
Comment (peterkeating) [inline content/Assets/Styles/settings/_form.scss]: Probably better if these were specified in `rem`. Favour `rem` over `px` when defining variables.

File: content/Assets/Styles/components/form/_radio.scss [hunk 4]
```diff
@@ -0,0 +1,90 @@
...
+            /* Show focus colour */
+            &:focus + .radios__label::before {           
+                @include focus-form-element;
+            }
+        }
+
+        &__label {
+            display: inline-block;
+            margin-bottom: 0;
+            padding-left: $form-radio-label-padding-left;
```
Comment (peterkeating) [inline content/Assets/Styles/components/form/_radio.scss]: Use `@include rem`

**Comment by bengammon** [comment]
Thanks for the feedback! Those points have now been addressed.

**Comment by bengammon** [comment]
@peterkeating Using this in the wild I have found a couple of things that we could do with adding to this before merging:

- [ ] Disabled button styles that are applied when `<button disabled>` is used
- [ ] Styles for a `.required` class for indicating which fields need to be filled in

I'll add to this list anything else I find is missing.

**Comment by peterkeating** [comment]
@bengammon Wasn't sure if you were going to handle validation within another PR. If this PR can essentially close https://github.com/EtchUK/Etch.OrchardCore.ThemeBoilerplate/issues/75 by completing all the items in the checklist, that would be great.

**Comment by bengammon** [comment]
@peterkeating I've just pushed more changes that complete the two checklist items I added for this issue.

I'm thinking of removing the validation point from #75 and creating a new issue for it? That way we can close #75 and keep any new conversations focused on validation?

**Comment by peterkeating** [comment]
@bengammon Sounds good, go ahead and create the issue and update the existing 👍

**Comment by bengammon** [comment]
@peterkeating All done pal

**Pull request merged**